src/main/java/org/homework/TodoService.java

```diff
@@ -0,0 +1,37 @@
+package org.homework;
+
+import java.util.Optional;
+
+public class TodoService {
+    private final TodoRepository repository;
+
+    public TodoService(TodoRepository repository) {
+        this.repository = repository;
+    }
+
+    public void addTodo(int id, String description) {
+        Todo todo = new Todo(id, description);
+        repository.save(todo);
+    }
+
+    public Optional<Todo> viewTodoById(int id) {
+        return repository.findById(id);
+    }
+
+    public boolean deleteTodoById(int id) {
+        if (repository.findById(id).isPresent()) {
+            repository.deleteById(id);
+            return true;
+        }
+        return false;
+    }
```

Comment:
```suggestion
   public boolean deleteTodoById(int id) {
    return repository.findById(id)
                     .map(todo -> {
                         repository.deleteById(id);
                         return true;
                     })
                     .orElse(false);
}
```

이런식으로 함수형 프로그래밍 형태로도 작성이 가능하니, 참고해주세요~
(아래 markTodoAsCompleted 메소드도 동일합니다.)